### DISCUSSION TITLE
How to calculate the uncertainty of the total energy calculated by MOOSE?

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#21814] [Q&A Modules: Phase field] [answered]
Hi everyone, 

How can I calculate the uncertainty of the total energy calculated by MOOSE? 
In my MOOSE input files, I used the following commands to calculate the total energy.
![image](https://user-images.githubusercontent.com/63679626/183783092-449a30bb-970a-452b-a678-f07a49b7d986.png)
Is there a function in MOOSE that can automatically calculate the uncertainty of the total energy?

Thank you.
Best,

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can do parametric studies on your uncertain parameters then examine the free energy using the stochastic module

Guillaume
